ambry-frontend/src/main/java/com/github/ambry/frontend/TtlUpdateHandler.java

```diff
@@ -126,38 +141,59 @@ private Callback<Void> securityProcessRequestCallback() {
     }
 
     /**
      * After {@link SecurityService#postProcessRequest} finishes, call {@link Router#updateBlobTtl} to update the TTL of
      * the blob in the storage layer.
-     * @param blobId the {@link BlobId} to update
+     * @param convertedBlobId the {@link BlobId} to update
      * @return a {@link Callback} to be used with {@link SecurityService#postProcessRequest}.
      */
-    private Callback<Void> securityPostProcessRequestCallback(BlobId blobId) {
+    private Callback<Void> securityPostProcessRequestCallback(BlobId convertedBlobId) {
       return buildCallback(metrics.updateBlobTtlSecurityPostProcessRequestMetrics, result -> {
         String serviceId = RestUtils.getHeader(restRequest.getArgs(), RestUtils.Headers.SERVICE_ID, true);
-        router.updateBlobTtl(blobId.getID(), serviceId, Utils.Infinite_Time, routerCallback(),
+        router.updateBlobTtl(convertedBlobId.getID(), serviceId, Utils.Infinite_Time, routerCallback(convertedBlobId),
             QuotaUtils.buildQuotaChargeCallback(restRequest, quotaManager, false));
       }, restRequest.getUri(), LOGGER, finalCallback);
     }
 
     /**
      * After {@link Router#updateBlobTtl} finishes, call {@link SecurityService#processResponse}.
+     * @param convertedBlobId the {@link BlobId} to update
      * @return a {@link Callback} to be used with {@link Router#updateBlobTtl}.
      */
-    private Callback<Void> routerCallback() {
+    private Callback<Void> routerCallback(BlobId convertedBlobId) {
       return buildCallback(metrics.updateBlobTtlRouterMetrics, result -> {
-        LOGGER.debug("Updated TTL of {}", RestUtils.getHeader(restRequest.getArgs(), RestUtils.Headers.BLOB_ID, true));
-        restResponseChannel.setHeader(RestUtils.Headers.DATE, new GregorianCalendar().getTime());
-        restResponseChannel.setHeader(RestUtils.Headers.CONTENT_LENGTH, 0);
-        securityService.processResponse(restRequest, restResponseChannel, null, securityProcessResponseCallback());
+        if (RequestPath.matchesOperation(blobIdStr, Operations.NAMED_BLOB)) {
+          // Set the named blob state to be 'READY' after the Ttl update succeed
+          if (!restRequest.getArgs().containsKey(NAMED_BLOB_VERSION)) {
+            throw new RestServiceException(
+                "Internal key " + NAMED_BLOB_VERSION + " is required in Named Blob TTL update callback!",
+                RestServiceErrorCode.InternalServerError);
+          }
+          long namedBlobVersion = (long) restRequest.getArgs().get(NAMED_BLOB_VERSION);
+          NamedBlobPath namedBlobPath = NamedBlobPath.parse(blobIdStr, restRequest.getArgs());
+          NamedBlobRecord record = new NamedBlobRecord(namedBlobPath.getAccountName(), namedBlobPath.getContainerName(),
+              namedBlobPath.getBlobName(), convertedBlobId.getID(), Utils.Infinite_Time, namedBlobVersion);
+          namedBlobDb.updateBlobTtlAndStateToReady(record).thenRun(() -> {
```

Comment:
nit: there is actually a utility method that deals with this type of conversion. It's in the CallbackUtils. You can do 
```
CallbackUtils.callCallbackAfter(namedBlobDb.updateBlobTtlAndStateToReady(record), (r, e) -> {
    if (e != null) {
        finalCallback.onComplete(null, e);
    } else {
        try {
            if (RestUtils.isDatasetVersionQueryEnabled(restRequest.getArgs())) {
                updateTtlForDatasetVersion();
            }
            processResponseHelper();
        } catch (RestServiceException ex) {
            finalCallback.onCompletion(null, ex);
        }
    }
})
```